crates/aptos-crypto/src/secp256k1.rs

```diff
@@ -0,0 +1,300 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides APIs for private keys and public keys used in SecP256K1.
+
+use crate::{
+    hash::{CryptoHash, HashValue},
+    traits,
+    traits::{CryptoMaterialError, ValidCryptoMaterial, ValidCryptoMaterialStringExt},
+};
+use anyhow::{anyhow, Result};
+use aptos_crypto_derive::{DeserializeKey, SerializeKey, SilentDebug, SilentDisplay};
+use core::convert::TryFrom;
+use serde::Serialize;
+
+/// libsecp256k1 expects pre-hashed messages of 32-bytes.
+pub const MESSAGE_LENGTH: usize = 32;
+/// SecP256K1 private keys are 256-bit.
+pub const PRIVATE_KEY_LENGTH: usize = 32;
+/// SecP256K1 public keys contain a prefix indicating compression and two 32-byte coordinates.
+pub const PUBLIC_KEY_LENGTH: usize = 65;
+/// SecP256K1 signatures are 256-bit.
+pub const SIGNATURE_LENGTH: usize = 64;
+
+/// SecP256K1 private key
+#[derive(DeserializeKey, Eq, PartialEq, SerializeKey, SilentDebug, SilentDisplay)]
+pub struct PrivateKey(pub(crate) libsecp256k1::SecretKey);
+
+#[cfg(feature = "assert-private-keys-not-cloneable")]
+static_assertions::assert_not_impl_any!(PrivateKey: Clone);
+
+#[cfg(any(test, feature = "cloneable-private-keys"))]
+impl Clone for PrivateKey {
+    fn clone(&self) -> Self {
+        let serialized: &[u8] = &(self.to_bytes());
+        PrivateKey::try_from(serialized).unwrap()
+    }
+}
+
+impl PrivateKey {
+    /// Serialize the private key into a byte vector
+    pub fn to_bytes(&self) -> Vec<u8> {
+        self.0.serialize().to_vec()
+    }
+
+    fn sign(&self, message: &libsecp256k1::Message) -> Signature {
+        let (signature, _recovery_id) = libsecp256k1::sign(message, &self.0);
```

Comment:
You never "normalize" the signature to have low `s`, do you? I thought you said you do.

This must also be done here.

Reply:
The library does that.

Reply:
Oh. `libsecp256k1` does it...

Reply:
Can you add a comment at the top of this function saying the signature is "normalized" to have "low" `s`, please? Otherwise, some of us will get a heart-attack every time we look at this function.